adapters/repos/db/index_queue.go

```diff
@@ -262,9 +267,14 @@ func (q *IndexQueue) indexer() {
 
 func (q *IndexQueue) pushToWorkers() error {
 	chunks := q.queue.borrowAllChunks()
-	for _, c := range chunks {
+	for i, c := range chunks {
 		select {
 		case <-q.ctx.Done():
+			// release unsent borrowed chunks
+			for _, c := range chunks[i:] {
+				q.queue.releaseChunk(c)
+			}
+
 			return errors.New("index queue closed")
```

Comment:
Does this need to be an error as shutting down Weaviate while the index in `INDEXING` state will trigger this.